components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -45,7 +48,42 @@ func (s *OIDCService) CreateClientConfig(ctx context.Context, req *connect.Reque
 		return nil, err
 	}
 
-	return nil, connect.NewError(connect.CodeUnimplemented, errors.New("gitpod.experimental.v1.OIDCService.CreateClientConfig is not implemented"))
+	result, err := s.oidcService.CreateClientConfig(ctx, &iam.CreateClientConfigRequest{
```

Comment:
Currently, who is part of the experiment, will be able to create such a config. For now, that feels acceptable, but we should add a check for the user also being a member of the team for which they are creating it.

Currently, the config is not attached to any entity (team/org). This will require that the TeamID is specified in the incoming request, valdiated, validated for membership and then associated with the config.

Reply:
1. IC
2. let's make it happen in https://github.com/gitpod-io/gitpod/issues/15708 then